qcengine/programs/aimnet2.py

```diff
@@ -0,0 +1,127 @@
+        "thread_safe": True,
+        "thread_parallel": False,
+        "node_parallel": False,
+        "managed_memory": False,
+    }
+
+    version_cache: Dict[str, str] = {}
+
+    @staticmethod
+    def found(raise_error: bool = False) -> bool:
+        return which_import(
+            "pyaimnet2",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install via `pip install git+https://github.com/jthorton/AIMNet2.git@main`",
+        )
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        which_prog = which_import("pyaimnet2")
+        if which_prog not in self.version_cache:
+            import pyaimnet2
+
+            self.version_cache[which_prog] = safe_version(pyaimnet2.__version__)
+
+        return self.version_cache[which_prog]
+
+    def load_model(self, name: str):
+        model_name = name.lower()
+        if model_name in self._CACHE:
+            return self._CACHE[model_name]
+
+        from pyaimnet2 import load_model
+
+        model = load_model(model_name=model_name)
+        self._CACHE[model_name] = model
+        return self._CACHE[model_name]
+
+    def compute(self, input_data: "AtomicInput", config: "TaskConfig"):
+        self.found(raise_error=True)
+        import torch
+        from qcengine.units import ureg
+
+        # check we can run on the set of elements
+        known_elements = {"H", "B", "C", "N", "O", "F", "Si", "P", "S", "Cl", "As", "Se", "Br", "I"}
+        target_elements = set(input_data.molecule.symbols)
+
+        unknown_elements = target_elements - known_elements
+        if unknown_elements:
+            raise InputError(f"AIMNET2 model {input_data.model.method} does not support elements {unknown_elements}.")
+
+        method = input_data.model.method
+        # load the model using the method as the file name
+        model = self.load_model(name=method)
+
+        # build the required input data
+        aimnet_input = {
+            "coord": torch.tensor(
+                [input_data.molecule.geometry * ureg.conversion_factor("bohr", "angstrom")],
+                dtype=torch.float64,
+                device="cpu",
+            ),
+            "numbers": torch.tensor([input_data.molecule.atomic_numbers], dtype=torch.long, device="cpu"),
+            "charge": torch.tensor([input_data.molecule.molecular_charge], dtype=torch.float64, device="cpu"),
+        }
+
+        if input_data.driver == "gradient":
+            aimnet_input["coord"].requires_grad_(True)
+        out = model(aimnet_input)
+
+        ret_data = {
+            "success": False,
+            "properties": {
+                "return_energy": out["energy"].item() * ureg.conversion_factor("eV", "hartree"),
+                "return_gradient": (
+                    -1.0 * out["forces"][0].detach().numpy() * ureg.conversion_factor("eV / angstrom", "hartree / bohr")
+                )
+                .ravel()
+                .tolist(),
```

Comment:
I'm pretty sure you can leave the value as a shaped numpy array. the AtomicResult() creation will handle the rest.